modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -1340,3 +1324,143 @@ export async function signRecoveryMpcV2(
+  };
+}
+
+/**
+ * Retrieves the MPC v2 recovery key shares from the provided user and backup key shares.
+ *
+ * @param {string} encryptedMPCv2UserKey
+ * @param {string} encryptedMPCv2BackupKey
+ * @param {string} [walletPassphrase] - The passphrase used to decrypt the key shares
+ * @returns {Promise<{ userKeyShare: KeyShare, backupKeyShare: KeyShare, commonKeyChain: string }>}
+ *
+ * @async
+ */
+async function getMpcV2RecoveryKeySharesFromReducedKey(
+  encryptedMPCv2UserKey: string,
+  encryptedMPCv2BackupKey: string,
+  walletPassphrase?: string
+): Promise<{
+  userKeyShare: Buffer;
+  backupKeyShare: Buffer;
+  commonKeyChain: string;
+}> {
+  const userCompressedPrv = Buffer.from(sjcl.decrypt(walletPassphrase, encryptedMPCv2UserKey), 'base64');
+  const bakcupCompressedPrv = Buffer.from(sjcl.decrypt(walletPassphrase, encryptedMPCv2BackupKey), 'base64');
+
+  const userPrvJSON: DklsTypes.ReducedKeyShare = DklsTypes.getDecodedReducedKeyShare(userCompressedPrv);
+  const backupPrvJSON: DklsTypes.ReducedKeyShare = DklsTypes.getDecodedReducedKeyShare(bakcupCompressedPrv);
+  const userKeyRetrofit: DklsTypes.RetrofitData = {
+    xShare: {
+      x: Buffer.from(userPrvJSON.prv).toString('hex'),
+      y: Buffer.from(userPrvJSON.pub).toString('hex'),
+      chaincode: Buffer.from(userPrvJSON.rootChainCode).toString('hex'),
+    },
+    xiList: userPrvJSON.xList.slice(0, 2),
+  };
+  const backupKeyRetrofit: DklsTypes.RetrofitData = {
+    xShare: {
+      x: Buffer.from(backupPrvJSON.prv).toString('hex'),
+      y: Buffer.from(backupPrvJSON.pub).toString('hex'),
+      chaincode: Buffer.from(backupPrvJSON.rootChainCode).toString('hex'),
+    },
+    xiList: backupPrvJSON.xList.slice(0, 2),
+  };
+  const [user, backup] = await DklsUtils.generate2of2KeyShares(userKeyRetrofit, backupKeyRetrofit);
+  const userKeyShare = user.getKeyShare();
+  const backupKeyShare = backup.getKeyShare();
+  const commonKeyChain = DklsTypes.getCommonKeychain(userKeyShare);
+  return { userKeyShare, backupKeyShare, commonKeyChain };
+}
+
+/**
+ * Gets the combined key for GG18
+ * @param encryptedGG18UserKey encrypted GG18 user key
+ * @param encryptedGG18BackupKey encrypted GG18 backup key
+ * @param walletPassphrase wallet passphrase
+ * @returns key shares
+ */
+function getKeyCombinedFromTssKeyShares(
+  encryptedGG18UserKey: string,
+  encryptedGG18BackupKey: string,
+  walletPassphrase?: string
+): [ECDSAMethodTypes.KeyCombined, ECDSAMethodTypes.KeyCombined] {
+  let backupPrv;
+  let userPrv;
+  try {
+    backupPrv = sjcl.decrypt(walletPassphrase, encryptedGG18BackupKey);
+    userPrv = sjcl.decrypt(walletPassphrase, encryptedGG18UserKey);
+  } catch (e) {
+    throw new Error(`Error decrypting backup keychain: ${e.message}`);
+  }
+
+  const userSigningMaterial = JSON.parse(userPrv) as ECDSAMethodTypes.SigningMaterial;
+  const backupSigningMaterial = JSON.parse(backupPrv) as ECDSAMethodTypes.SigningMaterial;
```

Comment:
can you not use io-ts to safely parse these?

Reply:
good point, we could. I'd rather do this in a separate PR though, this PR is primarily moving around the same functions without changing their behaviour too much.